src/user/controller/password.ts

```diff
@@ -3,9 +3,35 @@ import { Context } from '@curveball/core';
 import * as UserService from '../service';
 import { Forbidden } from '@curveball/http-errors';
 import * as privilegeService from '../../privilege/service';
+import log from '../../log/service';
+import { EventType } from '../../log/types';
 
 class UserPasswordController extends Controller {
 
+  async post(ctx: Context<any>) {
+
+    if (!await privilegeService.hasPrivilege(ctx, 'admin')) {
+      throw new Forbidden('Only users with the "admin" privilege may create new password');
+    }
+
+    const user = await UserService.findById(parseInt(ctx.params.id, 10));
+    const userNewPassword = ctx.request.body.newPassword;
+
+    if (user.type !== 'user') {
+      throw new Forbidden('You can only create passwords for users');
+    }
+
+    await UserService.createPassword(user, userNewPassword);
+
+    ctx.session = {
+      user: user,
+    };
+    log(EventType.changePasswordSuccess, ctx);
```

Comment:
Should I create `EventType.createPasswordSuccess` for this?

Reply:
I don't really think this endpoint is needed. `PUT` can be used for creating or modifying, and we don't really want the client to have to decide whether it should be created or modified. The only thing really needed to fix this bug is to change the `UPDATE` query to `INSERT ... ON DUPLICATE KEY UPDATE`